V22 week2 fixes

jakobsj opened the pull request:
I updated notebooks 1, 2, 3 and 4 with year and syntax changes. I believe @gfardell already updated notebook 5 (laminography) elsewhere.  This PR may need to be pointed to a different branch, v22_all or something?

ccpi-admin committed:
Update 01

ccpi-admin committed:
Update 02 for v22

ccpi-admin committed:
Update notebook 3 PDHG

ccpi-admin committed:
update years

ccpi-admin committed:
Update notebook 4 SPDHG for v22

ccpi-admin committed:
Correct year notebook 2

gfardell commented on examples/2_Iterative/02_tikhonov_block_framework.ipynb:
```diff
@@ -27,7 +27,7 @@
     "#   limitations under the License.\n",
     "\n",
-    "#   Copyright 2019 UKRI-STFC, The University of Manchester, Technical University of Denmark\n",
+    "#   Copyright 2019-2022 UKRI-STFC, The University of Manchester, Technical University of Denmark\n",
```
I think this needs the new header, not just the date range.

gfardell commented on examples/2_Iterative/03_PDHG.ipynb:
```diff
@@ -718,8 +732,8 @@
     "plt.rcParams.update({'lines.linewidth': 5})\n",
-    "plt.plot(fbp_recon.subset(horizontal_y = int(ig2D.voxel_num_y/2)).as_array(), label = 'FBP')\n",
-    "plt.plot(pdhg_l1.solution.subset(horizontal_y = int(ig2D.voxel_num_y/2)).as_array(), label = 'L1 regularisation')\n",
+    "plt.plot(fbp_recon.get_slice(horizontal_y = int(ig2D.voxel_num_y/2)).as_array(), label = 'FBP')\n",
```
Can we use assign the row number to a variable as it's used many times:
`row_num = int(ig2D.voxel_num_y/2)`

plt.plot(fbp_recon.get_slice(horizontal_y =row_num).as_array(), label = 'FBP')\n

gfardell commented on examples/2_Iterative/04_SPDHG.ipynb:
```diff
@@ -23,7 +23,7 @@
     "#   limitations under the License.\n",
     "\n",
-    "#   Copyright 2019 UKRI-STFC, University of Bath\n",
+    "#   Copyright 2019-2022 UKRI-STFC, University of Bath\n",
```
header needs updated text

gfardell commented on examples/2_Iterative/04_SPDHG.ipynb:
```diff
@@ -153,7 +153,11 @@
+    "\n",
+    "The data set can be downloaded here, after which you will need to unzip the file and update `pathname` below:\n",
+    "\n",
+    "https://zenodo.org/record/4822516/files/walnut.zip?download=1"
```
You can remove '?download=1'

gfardell reviewed (approved):
It's really hard to check these thoroughly through github. I would suggest we spend an hour together to go through them all (5mins per notebook) so we can keep a level of consistency.

gfardell committed:
V22 week2 additional suggestions (#96)

* Checked notebooks
fixed headers
fixed latex rendereding with vscode
fixed section formatting
added excercise snippits
fixed links between notebooks/sections

* fixed PDHG link text and examples

* rename TomoPhantom to cilTomoPhantom

Co-authored-by: lauramurgatroyd <laura.murgatroyd@stfc.ac.uk>
